ast.c

```diff
@@ -202,6 +203,10 @@ static VALUE
 node_id_for_backtrace_location(rb_execution_context_t *ec, VALUE module, VALUE location)
 {
     int node_id;
+
+    if (!rb_obj_is_kind_of(location, rb_cBacktraceLocation)) {
```

Comment:
```suggestion
    if (!rb_frame_info_p(location)) {
```
as same as `ast_s_of`.